src/utils/IdealGraphVisualizer/Coordinator/src/main/java/com/sun/hotspot/igv/coordinator/actions/GraphRemoveCookie.java

```diff
@@ -34,6 +40,28 @@ public GraphRemoveCookie(InputGraph graph) {
 
     @Override
     public void remove() {
+        List<InputGraph> list = graph.getGroup().getGraphs();
+        WindowManager manager = WindowManager.getDefault();
+        for (Mode m : manager.getModes()) {
+            for (TopComponent t : manager.getOpenedTopComponents(m)) {
+                if (t instanceof EditorTopComponent) {
+                    DiagramViewModel model = ((EditorTopComponent) t).getModel();
+                    if (!model.getGroup().getGraphs().contains(graph)) {
+                        continue;
+                    }
+                    int firstPosition = model.getFirstPosition();
+                    int secondPosition = model.getSecondPosition();
+                    int targetPosition = list.indexOf(graph);
+                    if (targetPosition < firstPosition) {
+                        firstPosition--;
+                    }
+                    if (targetPosition < secondPosition) {
+                        secondPosition--;
+                    }
```

Comment:
```suggestion
                    }
                    if (targetPosition == firstPosition || targetPosition == secondPosition) {
                        t.close();
                    }
```

Reply:
Thank you for your suggestion. I've put this code before decrementing.

Reply:
sure. thanks!